apps/institution-service/src/controllers/institutionController.ts

```diff
@@ -180,11 +190,80 @@ const getPaginationOptions = (req: Request): PaginationOptions => {
   return { page, limit, offset };
 };
 
+const whereAggregatorIntegrationConditions = (req: Request) => {
+  const {
+    supportsIdentification,
+    supportsAggregation,
+    supportsHistory,
+    supportsVerification,
+    supportsOauth,
+    isActive,
+  } = req.query;
+  const whereConditions: WhereConditions = {};
+  if (supportsIdentification === "true") {
+    whereConditions["supports_identification"] = true;
+  }
+  if (supportsAggregation === "true") {
+    whereConditions["supports_aggregation"] = true;
+  }
+  if (supportsHistory === "true") {
+    whereConditions["supports_history"] = true;
+  }
+  if (supportsVerification === "true") {
+    whereConditions["supports_verification"] = true;
+  }
+  if (supportsOauth === "true") {
+    whereConditions["supports_oauth"] = true;
+  }
+  if (isActive) {
+    whereConditions["isActive"] = isActive === "true" ? true : false;
+  }
+
+  return whereConditions;
+};
+
+const whereInstitutionConditions = (req: Request): WhereConditions => {
+  const { search, aggregatorName } = req.query;
+
+  const whereConditions: WhereConditions = {};
+
+  if (search) {
+    whereConditions[Op.or] = [
+      { name: { [Op.iLike]: `%${search as string}%` } },
+      { keywords: { [Op.contains]: [search as string] } },
```

Comment:
It's a bit trickier to do LIKE on an array but it's doable. It might need to be done in raw sql. This is currently case sensitive and has to be an exact match so it's not the best. I'm probably going to have to rewrite most of the sql query so I'll see if I can fix this while I'm fixing everything else.